### PR TITLE
Add Object.assign polyfill support

### PR DESCRIPTION
**Pull request by emlove**
This PR adds a new js module compatibility.js, which currently adds polyfill support for Object.assign.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,11 +15,13 @@
     "frontend_html": "script/vulcanize.js",
     "frontend_prod": "npm run js_prod && npm run frontend_html",
     "frontend_demo": "npm run js_demo && npm run frontend_html",
-    "ru_all": "npm run ru_core && npm run ru_demo",
+    "ru_all": "npm run ru_core && npm run ru_compatibility && npm run ru_demo",
     "ru_core": "rollup --config rollup/core.js",
+    "ru_compatibility": "rollup --config rollup/compatibility.js",
     "ru_demo": "rollup --config rollup/demo.js",
-    "watch_ru_all": "npm run watch_ru_core && npm run watch_ru_demo",
+    "watch_ru_all": "(npm run watch_ru_core & npm run watch_ru_compatibility & npm run watch_ru_demo) && wait",
```
Comment (armills) [inline package.json]: These were being run in sequence previously, which ends up only watching core. I believe the intention is to watch all of them at once.

Comment (balloob) [inline package.json]: interesting, I never realized that. Then again, we hardly develop anything but the core.

File: src/compatibility.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+require('es6-object-assign').polyfill();
```
Comment (balloob) [inline src/compatibility.js]: You can make this ES6.

```js
import objAssign from es6-object-assign;
objAssign.polyfill();
```

File: src/compatibility.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
...
+module.exports = Object;
```
Comment (balloob) [inline src/compatibility.js]: You don't need an export.

File: rollup/compatibility.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+export default Object.assign({}, config, {
+  entry: 'src/compatibility.js',
+  targets: [
+    {
+        dest: 'build/compatibility.js',
+        format: 'iife',
+        moduleName: 'compatibility',
+    },
+  ],
+  exports: 'default',
```
Comment (balloob) [inline rollup/compatibility.js]: No need to expose an export as this is a polyfill.

File: package.json (hunk 1 above)
Comment (armills) [inline package.json]: That's what I figured. Just wanted to confirm.

File: src/compatibility.js (hunk 2 above)
Comment (armills) [inline src/compatibility.js]: Ah, this is what I was missing.

**Pull request merged**